app/webserver/helpers.go

```diff
@@ -240,36 +240,42 @@ func joinGroup(joinGroupmessage JoinGroupMessage) *store.Session {
 	log.Infoln("[axolotl] joinGroup", joinGroupmessage.ID)
 	group, err := textsecure.JoinGroup(joinGroupmessage.ID)
 	if err != nil {
+		log.Errorln("[axolotl] joinGroup failed", err)
 		// update group also in the case that the group is already joined to remove the join message
-		if group != nil {
-			session, _ := store.SessionsModel.GetByUUID(group.Hexid)
-			session.Name = group.DecryptedGroup.Title
-			session.GroupJoinStatus = group.JoinStatus
-			store.UpdateSession(session)
+		if group == nil {
+			return nil
 		}
-		log.Errorln("[axolotl] joinGroup", err)
-		ShowError(err.Error())
-		return nil
+
 	}
+
 	log.Infoln("[axolotl] joining group was succesful", group.Hexid)
 	members := ""
-	for _, member := range group.DecryptedGroup.Members {
-		members = members + string(member.Uuid) + ","
+	// members cannot be read if the group is not yet joined
+	if group.JoinStatus == store.GroupJoinStatusJoined {
+		for _, member := range group.DecryptedGroup.Members {
+			members = members + string(member.Uuid) + ","
+		}
 	}
```

Comment:
When is `group.JoinStatus == store.GroupJoinStatusJoined`? And is it correct that in all other cases members should be `""`?
I want to be sure that we don't delete members of groups in unhappy cases.

Reply:
`store.GroupJoinStatusJoined` is a constant with value `0` and this comparison indicates, that we are part of the group. When this is not true, we cannot decrypt the group, meaning, we don't know who is a member, so in this case the members are empty. Otherwise we know who invited us and we can set this person as member.